### ISSUE TITLE
Containers: Updating the image will not run setup which can cause issues if updates require infrastructure modifications

### ISSUE THREAD

**Issue by ramonsmits** [Particular/ServiceControl#4300]
### Is your improvement related to a problem? Please describe.

Updating container images to a newer version via the following very common pattern can cause issues if the update(s) require infrastructure modifications:

```
docker-compose pull
docker-compose up --detach 
```

Lets say a customer uses the tag `particular/servicecontrol:5` and wants to update the the latest minor/patch would run the above and expects the container to just be updated.

Instead, the customer needs to:
1. Pull
2. Run setup
3. Run normal

Currently the image does not have any validation if the container version is aligned with storage or infrastructure and does not prevent startup and can cause errors which might be difficult to troubleshoot by users.

### Always run setup at start

Just always run setup at start. This will impact time to first message but require setup and run to share the same credentials.

### Detect version upgrade and hint to run setup

For environments to run least privilege with different connection strings we could at minimum have a validation that checks if the current image version (app version) matches with the version of the last time setup was run and when no last setup version exist or there is a mismatch not start the image.

1. Setup write version to file
2. Normal run reads file with version
        
#### Only run setup when detecting version upgrade

Detect version upgrade based on a local (file) state and only run a migration when there is a version mismatch.

**Comment by ramonsmits** [comment]
## Workaround: Always run setup at container start:

Mount a custom entrypoint bash that will run setup and regular start

Example for ServiceControl image:
```sh
#! /bin/bash
./app/ServiceControl --setup
./app/ServiceControl
```

See the following for a full example: 

- https://github.com/ramonsmits/ServiceControl-Docker/tree/master

**Comment by DavidBoike** [comment]
Implemented in [ServiceControl 5.7.0](https://github.com/Particular/ServiceControl/releases/5.7.0) using the `--setup-and-run` argument.

See docs:

* [ServiceControl instances](https://docs.particular.net/servicecontrol/servicecontrol-instances/deployment/containers#initial-setup-simplified-setup)
* [Audit instances](https://docs.particular.net/servicecontrol/audit-instances/deployment/containers#initial-setup-simplified-setup)
* [Monitoring instances](https://docs.particular.net/servicecontrol/monitoring-instances/deployment/containers#initial-setup-simplified-setup)
